equinor/EchoFramework
# feat: Load module only when on its path

### Description

We only want to start the process of loading a module when we are on that module's path. This is a suggestion of how we can achieve that. The mediator here is what starts the module loading process by calling `startLoadingModules`. We provide it our current path, which in turn will use it to only load a module if we're on its path.

### Remarks

This is only a suggestion, and maybe a step in the right direction. We might want to look more into unloading modules as well, but this could be more of an MVP.

## How to test
You need both this branch, and the [related branch](https://github.com/equinor/EchoCore/pull/83) in echo-base for this to work.

In echo-base do:

```
npm run build
yalc publish
```
In echo-framework do:
```
yalc add @equinor/echo-base
npm install
npm run build
yalc publish
```
In EchopediaWeb do:
```
yalc add @equinor/echo-framework
yalc add @equinor/echo-base
npm install
```
If you get an error when trying to start the application, try to copy the `dist` folder from echo-framework, and place it under the folder `.yalc/@equinor/echo-framework` in echopedia-web.

Now try to start again.

When the app has started, open the network tab, and in the filter write "echo3d". The echoModuleManifest.json should be fetched on startup, but the index.js should not. If you open echo3dWeb, you should see that the index.js file is fetched.

## Related PRs

https://github.com/equinor/EchoCore/pull/83
